src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp

```diff
@@ -393,12 +394,15 @@ class ShenandoahCardCluster: public CHeapObj<mtGC> {
   static const uint16_t ObjectStartsInCardRegion = 0x80;
   static const uint16_t FirstStartBits           = 0x7f;
 
+  // Check that we have enough bits to store the largest possible offset into a card for an object start
+  static const int MaxCardSize = NOT_LP64(512) LP64_ONLY(1024);
+  STATIC_ASSERT((MaxCardSize / HeapWordSize) - 1 <= FirstStartBits);
+
```

Comment:
May be leave a reference to the raw values for `MaxCardSize` as coming from the range for `GCCardSizeInBytes` e.g. in `gc_globals.hpp`.